src/test/ut/ut_cortxfs_dir_ops.c

```diff
@@ -871,34 +871,33 @@ static void create_remove_subdir(void **state)
 	struct ut_cfs_params *ut_cfs_obj = ENV_FROM_STATE(state);
 	cfs_ino_t *pinode =  &ut_cfs_obj->file_inode;
 	cfs_ino_t cinode;
-	struct stat before_create;
-	struct stat after_create;
-	struct stat after_rmdir;
+	struct cfs_fh *parent_fh = NULL;
+	struct stat *before_create = NULL;
+	struct stat *after_create = NULL;
+	struct stat *after_rmdir = NULL;
 
-	memset(&before_create, 0, sizeof(before_create));
-	rc = cfs_getattr(ut_cfs_obj->cfs_fs, &ut_cfs_obj->cred,
-			 pinode, &before_create);
+	rc = cfs_fh_from_ino(ut_cfs_obj->cfs_fs, pinode, &parent_fh);
 	ut_assert_int_equal(rc, 0);
+	before_create = cfs_fh_stat(parent_fh);
 
 	rc = cfs_mkdir(ut_cfs_obj->cfs_fs, &ut_cfs_obj->cred,
 		       pinode, "childdir", 0755, &cinode);
 	ut_assert_int_equal(rc, 0);
 
-	memset(&after_create, 0, sizeof(after_create));
-	rc = cfs_getattr(ut_cfs_obj->cfs_fs, &ut_cfs_obj->cred,
-			 pinode, &after_create);
+	rc = cfs_fh_from_ino(ut_cfs_obj->cfs_fs, pinode, &parent_fh);
```

Comment:
Here you are again over writing the `parent_fh` pointer without destroying it hence memory leak. Things to remember if we destroy FH then stat reference from that won't be valid after destruction. So you have to memcpy stat from fh instead of getting weak reference from it.